libtenzir/builtins/operators/enumerate.cpp

```diff
@@ -114,12 +116,23 @@ class enumerate_operator final : public crtp_operator<enumerate_operator> {
   std::string field_;
 };
 
-class plugin final : public virtual operator_plugin<enumerate_operator> {
+class plugin final : public virtual operator_plugin<enumerate_operator>,
+                     public virtual operator_factory_plugin {
 public:
   auto signature() const -> operator_signature override {
     return {.transformation = true};
   }
 
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    std::string field;
```

Comment:
This needs to be an expression.

Reply:
This is the name of the output field, which might not exist already. How would an ast::expression resolve here?